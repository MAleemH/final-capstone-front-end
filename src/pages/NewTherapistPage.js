/* eslint-disable no-console */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import '../css/NewTherapistPage.css';
import { useDispatch } from 'react-redux';
import specializationArr from '../components/Speciliazation';
import backImg from '../img/back.png';
import fadingBreak from '../img/fading_break.png';
import logoImg from '../img/logo.png';
import avatarImg from '../img/user.png';
import { postTherapist, uploadTherapist } from '../redux/Therapy/therapySlice';

function NewTherapistPage() {
  const dispatch = useDispatch();
  const [uploadFile, setUploadFile] = useState('');
  const [inputName] = useState('');
  const [inputAddress] = useState('');
  // const [cloudinaryImage, setCloudinaryImage] = useState('');
  // const cloudinaryAPI = 'https://api.cloudinary.com/v1_1/drhbncewu/image/upload';
  // const postDataAPI = 'http://your-database-api.com/upload';

  const handleUpload = async (e) => {
    e.preventDefault();
    const formData = new FormData();
    formData.append('file', uploadFile);
    formData.append('upload_preset', 'f00ugkxm');
    const secureUrl = await dispatch(uploadTherapist(formData));
    return secureUrl;
  };

  const handleSubmit = async () => {
    const cloudinaryImageUrl = await handleUpload();
    const dataToSend = {
      secureUrl: cloudinaryImageUrl,
      name: inputName,
      address: inputAddress,
      // Add other form data fields as needed
    };
    // Send the data to your database
    const response = await dispatch(postTherapist(dataToSend));
    console.log(response);
  };

  return (
    <div className="new_therapist_body">

      <header className="new_therapist_header">
        <nav>
          <Link to="/">
            <img src={backImg} alt="" />
          </Link>
        </nav>
      </header>

      <main className="new_therapist_main">
        <figure className="logo_fig">
          <Link to="/">
            <img src={logoImg} alt="" />
          </Link>
        </figure>

        <div className="fadebreak_box">
          <figure>
            {' '}
            <img src={fadingBreak} alt="" />
            {' '}
          </figure>
          <p> &nbsp; New Therapist &nbsp; </p>
          <figure>
            {' '}
            <img className="rotate_breakline" src={fadingBreak} alt="" />
            {' '}
          </figure>
        </div>

        <form action="" className="new_therapist_form">
          <fieldset className="fieldset_border_none">
            <input className="input_name" type="text" placeholder="Name" aria-label="Input Name" required />

            <input className="input_name" type="text" placeholder="Address (State and City)" aria-label="Input Address" required />

            <input className="input_name" type="phone" placeholder="Phone" aria-label="Input Phone" required />

            <input className="input_name" type="email" placeholder="Email" aria-label="Input Email" required />

            <input className="input_name" type="datetime-local" placeholder="Time" aria-label="Input Date and Time" required />

            <select aria-label="Input Label" className="input_name" id="specializationId">
              {specializationArr.map((specialty) => (
                <option key={specialty} value={specialty.value} aria-label="Input Specialization">{specialty.name}</option>
              ))}
            </select>

            <div className="input_name image_input_container">
              <input type="file" aria-label="Add Image" onChange={(event) => { setUploadFile(event.target.files[0]); }} />
              <img src={avatarImg} alt="Choose File" />
              <p>Add Image</p>
            </div>

          </fieldset>

          <fieldset className="fieldset_border_none new_therapist_action">
            <button type="button" onClick={handleSubmit}>Submit</button>
          </fieldset>
        </form>

      </main>

      <footer className="new_therapist_footer">
        <ul className="list_style_none">
          <li><Link to="/new_therapist" className="td_none"> terms </Link></li>
          <li><Link to="/new_therapist" className="td_none">  privacy </Link></li>
        </ul>
      </footer>

    </div>
  );
}

export default NewTherapistPage;
