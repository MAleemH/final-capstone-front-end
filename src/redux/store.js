import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
