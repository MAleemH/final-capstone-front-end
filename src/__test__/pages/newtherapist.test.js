import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import specializationArr from '../../components/speciliazation';
import NewTherapistPage from '../../pages/NewTherapistPage';

it('NewTherapistPage component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <NewTherapistPage specializationArr={specializationArr} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
