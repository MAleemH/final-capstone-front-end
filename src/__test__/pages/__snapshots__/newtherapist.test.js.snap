// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTherapistPage component renders correctly 1`] = `
<div
  className="new_therapist_body"
>
  <header
    className="new_therapist_header"
  >
    <nav>
      <button
        className="back_none"
        onClick={[Function]}
        type="button"
      >
        <img
          alt=""
          src="back.png"
        />
      </button>
    </nav>
  </header>
  <main
    className="new_therapist_main"
  >
    <figure
      className="logo_fig"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          src="logo.png"
        />
      </a>
    </figure>
    <div
      className="fadebreak_box"
    >
      <figure>
         
        <img
          alt=""
          src="fading_break.png"
        />
         
      </figure>
      <p>
         
        New Therapist
         
      </p>
      <figure>
         
        <img
          alt=""
          className="rotate_breakline"
          src="fading_break.png"
        />
         
      </figure>
    </div>
    <form
      action=""
      className="new_therapist_form"
    >
      <fieldset
        className="fieldset_border_none"
      >
        <input
          aria-label="Input Name"
          className="input_name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
        <input
          aria-label="Input Address"
          className="input_name"
          onChange={[Function]}
          placeholder="Address (State and City)"
          required={true}
          type="text"
          value=""
        />
        <input
          aria-label="Input Phone"
          className="input_name"
          onChange={[Function]}
          placeholder="Phone"
          required={true}
          type="phone"
        />
        <input
          aria-label="Input Email"
          className="input_name"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
        <select
          aria-label="Input Label"
          className="input_name"
          id="specializationId"
          onChange={[Function]}
        >
          <option
            aria-label="Input Specialization"
            value=""
          >
            Select Specialization
          </option>
          <option
            aria-label="Input Specialization"
            value="Cardiovascular"
          >
            Cardiovascular
          </option>
          <option
            aria-label="Input Specialization"
            value="Emotional"
          >
            Emotional
          </option>
          <option
            aria-label="Input Specialization"
            value="Romance"
          >
            Romance
          </option>
          <option
            aria-label="Input Specialization"
            value="Sexual"
          >
            Sexual
          </option>
          <option
            aria-label="Input Specialization"
            value="Academic"
          >
            Academic
          </option>
          <option
            aria-label="Input Specialization"
            value="Career"
          >
            Career
          </option>
          <option
            aria-label="Input Specialization"
            value="Marital"
          >
            Marital
          </option>
          <option
            aria-label="Input Specialization"
            value="Dating"
          >
            Dating
          </option>
        </select>
        <div
          className="input_name image_input_container"
        >
          <input
            aria-label="Add Image"
            onChange={[Function]}
            required={true}
            type="file"
          />
          <img
            alt="Choose File"
            src="user.png"
          />
          <p>
            Add Image
          </p>
        </div>
      </fieldset>
      <fieldset
        className="fieldset_border_none new_therapist_action"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </fieldset>
    </form>
  </main>
  <footer
    className="new_therapist_footer"
  >
    <ul
      className="list_style_none"
    >
      <li>
        <a
          className="td_none"
          href="/new_therapist"
          onClick={[Function]}
        >
           terms 
        </a>
      </li>
      <li>
        <a
          className="td_none"
          href="/new_therapist"
          onClick={[Function]}
        >
            privacy 
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
